diff.py

```diff
@@ -1980,73 +2079,70 @@ def flush_matching() -> None:
+            headers=(
+                Text("TARGET"),
+                Text(f"  CURRENT ({new_diff.score})"),
+                Text(f"  PREVIOUS ({old_diff.score})"),
+            ),
+            current_score=new_diff.score,
+            previous_score=old_diff.score,
         )
+        old_chunks = chunk_diff_lines(old_diff.lines)
+        new_chunks = chunk_diff_lines(new_diff.lines)
+        diff_lines = []
+        empty = OutputLine(Text(), Text(), None, True, None, None)
+        assert len(old_chunks) == len(new_chunks), "same target"
+        for old_chunk, new_chunk in zip(old_chunks, new_chunks):
+            if isinstance(old_chunk, list):
+                assert isinstance(new_chunk, list)
+                if not old_chunk and not new_chunk:
+                    # Most of the time lines sync up without insertions/deletions,
+                    # and there's no interdiffing to be done.
+                    continue
+                differ = difflib.SequenceMatcher(
+                    a=old_chunk, b=new_chunk, autojunk=False
+                )
+                for (tag, i1, i2, j1, j2) in differ.get_opcodes():
+                    if tag in ["equal", "replace"]:
+                        for i, j in zip(range(i1, i2), range(j1, j2)):
+                            diff_lines.append((empty, new_chunk[j], old_chunk[i]))
+                    if tag in ["insert", "replace"]:
+                        for j in range(j1 + i2 - i1, j2):
+                            diff_lines.append((empty, new_chunk[j], empty))
+                    if tag in ["delete", "replace"]:
+                        for i in range(i1 + j2 - j1, i2):
+                            diff_lines.append((empty, empty, old_chunk[i]))
+            else:
+                assert isinstance(new_chunk, OutputLine)
+                # old_chunk.base and new_chunk.base have the same text since
+                # both diffs are based on the same target, but they might
+                # differ in color. Use the new version.
+                diff_lines.append((new_chunk, new_chunk, old_chunk))
         diff_lines = [
-            (
-                (
-                    base,
-                    new.fmt2,
-                    old.fmt2 or Text("-") if old != new else Text(),
-                ),
-                new.boring,
-            )
-            for (base, old, new) in output
+            (base, new, old if old != new else empty) for base, new, old in diff_lines
         ]
     else:
-        header_line = (Text("TARGET"), Text(f"  CURRENT ({new_diff.score})"))
-        diff_lines = [
-            ((base, new.fmt2), new.boring)
-            for (base, old, new) in output
-            if base or new.key2 is not None
-        ]
+        meta = TableMetadata(
+            headers=(
+                Text("TARGET"),
+                Text(f"  CURRENT ({new_diff.score})"),
+            ),
+            current_score=new_diff.score,
+            previous_score=old_diff.score,
```

Comment:
```suggestion
            previous_score=None,
```
and make previous_score nullable?